src/main/java/seedu/address/logic/commands/EditTypeCommand.java

```diff
@@ -30,12 +31,15 @@ protected static Person createEditedPerson(Person personToEdit, EditPersonDescri
 
         Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        Telegram updatedTelegram = editPersonDescriptor.getTelegram()
+                .orElse(personToEdit.getTelegram().orElse(Telegram.EMPTY_TELEGRAM));
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
```

Comment:
Telegram works the same as Email and Address right? They are all Optional. But the implementations are different, maybe we can change the getter methods for Email and Address to return Optional as well.
```suggestion
        Telegram updatedTelegram = editPersonDescriptor.getTelegram()
                .orElse(personToEdit.getTelegram().orElse(Telegram.EMPTY_TELEGRAM));
        Email updatedEmail = editPersonDescriptor.getEmail()
                .orElse(personToEdit.getEmail().orElse(Email.EMPTY_EMAIL));
        Address updatedAddress = editPersonDescriptor.getAddress()
                .orElse(personToEdit.getAddress().orElse(Address.EMPTY_ADDRESS);
```

Reply:
After reading further, I think Telegram's get() method should be removed instead of changing implementation of Email and Address